### PR TITLE
test: Migrate http2 test to node:test

### PR DESCRIPTION
**Pull request by Rantoledo**
Hi, 

These changes are related to the issue https://github.com/fastify/fastify/issues/5555
Changed `/test/http2` tests to use `node:test` instead of `tap`.
Basically, followed the same guidelines as my previous PR #5714.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/http2/closing.test.js [hunk 1]
```diff
@@ -9,102 +9,94 @@ const { once } = require('node:events')
...
+        // happen or not
       })
       session.on('error', () => {
         // Nothing to do here,
         // we are not interested in this error that might
         // happen or not
-        t.end()
+        done()
       })
+      fastify.close()
```
Comment (dancastillo) [inline test/http2/closing.test.js:51]: this should be unneeded with being called in `t.after` [here](https://github.com/fastify/fastify/pull/5757/files#diff-fc24aac3bf47d34df40dc90bc04449121c7fb3e258f2949cc3d0f639b00de29cR27)

File: test/http2/closing.test.js [hunk 2]
```diff
@@ -9,102 +9,94 @@ const { once } = require('node:events')
...
+        t.assert.strictEqual(headers[':status'], 200)
+        done()
+        this.destroy()
+      }).on('error', () => {
         // Nothing to do here,
         // we are not interested in this error that might
         // happen or not
-        t.end()
       })
+      fastify.close()
```
Comment (dancastillo) [inline test/http2/closing.test.js:88]: this should be unneeded with being called in `t.after` above

File: test/http2/closing.test.js (hunk 1 above)
Comment (Rantoledo) [reply test/http2/closing.test.js:51]: Let me see... I need to think again about the two first tests.

Comment (dancastillo) [reply test/http2/closing.test.js:51]: Good call out sorry didnt look at test file

**Review by gurgunday — APPROVED** [review]

**Review by jsumners — APPROVED** [review]

**Pull request merged**